contracts/src/OffChainMessageContracts/ValidatorSetSig.sol

```diff
@@ -0,0 +1,123 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+import {
+    WarpMessage,
+    IWarpMessenger
+} from "@avalabs/subnet-evm-contracts@1.2.0/contracts/interfaces/IWarpMessenger.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts@4.8.1/security/ReentrancyGuard.sol";
+
+// Message format for the WarpMessage payload to be forwarded to the target contract
+struct ValidatorSetSigMessage {
+    address validatorSetSigAddress;
+    address targetContractAddress;
+    bytes32 targetBlockChainID;
+    uint256 nonce;
+    bytes txPayload;
+}
+
+/**
+ * @dev Contract that verifies that a set threshold of validators from a given blockchainID
+ * have signed an off-chain Warp message and forwards the payload to the destination contract specified in the payload.
+ * The threshold itself is set by the validator themselves in their Warp configs:
+ * https://github.com/ava-labs/subnet-evm/blob/master/precompile/contracts/warp/config.go#L50
+ *
+ * This is intended to be used for safe off-chain governance of enabled contracts by having the target contract be owned by
```

Comment:
This should apply to any general access control mechanism on chain right? For example access through roles. 

Would clarify documentation to be more general access control, and use Ownable as an example.